recipes/isa-l/all/conanfile.py

```diff
@@ -0,0 +1,76 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+
+import os
+
+required_conan_version = ">=1.33.0"
+
+class LibisalConan(ConanFile):
+    name = "isa-l"
+    description = "Intel's Intelligent Storage Acceleration Library"
+    license = "BSD-3"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/01org/isa-l"
+    topics = ("isa-l", "compression")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC" :True
+    }
+    build_requires = (
+        "libtool/2.4.6",
+        "nasm/2.15.05",
+    )
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def validate(self):
+        if self.settings.arch not in [ "x86", "x86_64", "armv8" ]:
+            raise ConanInvalidConfiguration("CPU Architecture not supported")
+        if self.settings.os in ["Macos", "Windows"]:
```

Comment:
might be worth doing the not in linux or freebsd, unless android and ios are support?